MTA-1289 adding new mta cli commands: Analyze and Transform (#792)
windup/windup-documentation: docs/topics/mta-cli-analyze.adoc

To run analysis on application source code, run:

[source,terminal,subs="attributes+"]
----
windup-cli analyze --input=<path/to/source/code> --output=<path/to/output/dir>
----

All flags:

[source,terminal,subs="attributes+"]
----
Analyze application source code

Usage:
  windup-cli analyze [flags]

Flags:
      --analyze-known-libraries   analyze known open-source libraries
  -h, --help                      help for analyze
  -i, --input string              path to application source code or a binary
      --json-output               create analysis and dependency output as json
      --list-sources              list rules for available migration sources
      --list-targets              list rules for available migration targets
  -m, --mode string               analysis mode. Must be one of 'full' or 'source-only' (default "full")
  -o, --output string             path to the directory for analysis output
      --rules stringArray         filename or directory containing rule files
      --skip-static-report        do not generate static report
  -s, --source stringArray        source technology to consider for analysis

>   -t, --target stringArray        target technology to consider for analysis

Reviewer: the `analyze` subcommand has some more options:

```
Flags:
  -l, --label-selector string     run rules based on specified label selector expression
      --maven-settings string     path to a custom maven settings file to use
      --overwrite                 overwrite output directory
      --skip-static-report        do not generate static report
```

Global Flags:
      --log-level int   log level (default 5)
----
.Usage example

. Get an example application to run analysis on
. List available target technologies
[source,terminal,subs="attributes+"]
----
windup-cli analyze --list-targets
----
. Run an analysis with a specified target technology
[source,terminal,subs="attributes+"]
----
windup-cli analyze --input=<path-to/example-applications/example-1> --output=<path-to-output-dir> --target=cloud-readiness
----
. Several analysis reports are created in your specified output path:
[source,terminal,subs="attributes+"]

----
$ ls ./output/ -1
analysis.log
dependencies.yaml
dependency.log
output.yaml
static-report
----

`output.yaml` is the file that contains the issues report.
`static-report` contains the static HTML report.

> `dependencies.yaml`contains a dependencies report.

Reviewer: There is a missing whitespace before "contains"

// Module included in the following assemblies:
//
// * docs/cli-guide/master.adoc

:_content-type: CONCEPT
[id="mta-cli-analyze_{context}"]
= Performing analysis using the command line

`Analyze` allows running source code and binary analysis using `analyzer-lsp`.

> To run analysis on application source code, run:

Reviewer 2: ```suggestion
To run analysis on application source code run the following command:
```
That's the way it's done in OADP

      --list-targets              list rules for available migration targets
  -l, --label-selector string     run rules based on specified label selector expression
      --maven-settings string     path to a custom maven settings file to use
      --overwrite                 overwrite output directory
      --skip-static-report        do not generate static report
  -m, --mode string               analysis mode. Must be one of 'full' or 'source-only' (default "full")
  -o, --output string             path to the directory for analysis output
      --rules stringArray         filename or directory containing rule files
      --skip-static-report        do not generate static report
  -s, --source stringArray        source technology to consider for analysis
  -t, --target stringArray        target technology to consider for analysis

Global Flags:
      --log-level uint32   log level (default 4)
      --no-cleanup         do not cleanup temporary resources
----
.Usage example

. Get an example application to run analysis on
. List available target technologies
[source,terminal,subs="attributes+"]
----
{mta-cli} analyze --list-targets
----

> . Run an analysis with a specified target technology

Reviewer 2: ```suggestion
. Run an analysis with a specified target technology.
```
I think you need a period at the end of these sentences.

[source,terminal,subs="attributes+"]
----
{mta-cli} analyze --input=<path-to/example-applications/example-1> --output=<path-to-output-dir> --target=cloud-readiness
----

> . Several analysis reports are created in your specified output path:

Reviewer 2: ```suggestion
. Several analysis reports are created in your specified output path:
```
In the preview file this looks like #1, shouldn't it be #3 ?

  -h, --help                      help for analyze
  -i, --input string              path to application source code or a binary
      --json-output               create analysis and dependency output as json
      --list-sources              list rules for available migration sources
      --list-targets              list rules for available migration targets
  -l, --label-selector string     run rules based on specified label selector expression
      --maven-settings string     path to a custom maven settings file to use
      --overwrite                 overwrite output directory
      --skip-static-report        do not generate static report
  -m, --mode string               analysis mode. Must be one of 'full' or 'source-only' (default "full")
  -o, --output string             path to the directory for analysis output
      --rules stringArray         filename or directory containing rule files
      --skip-static-report        do not generate static report
  -s, --source string             source technology to consider for analysis. To specify multiple sources, repeat the parameter: --source <source_1> --source <source_2> etc.

>   -t, --target string             target technology to consider for analysis. To specify multiple sources, repeat the parameter: --source <source_1> --source <source_2> etc.

Reviewer 3: ```suggestion
  -t, --target string             target technology to consider for analysis. To specify multiple targets, repeat the parameter: --target <target_1> --target <target_2> etc.
```

Reviewer 4: ^ do we need to fix our CLI output as well?

Global Flags:
      --log-level uint32   log level (default 4)
      --no-cleanup         do not cleanup temporary resources
----
.Usage example

. Get an example application to run analysis on.
. List available target technologies.
+
[source,terminal,subs="attributes+"]
----
{mta-cli} analyze --list-targets
----

> . Run an analysis with a specified target technology.

Reviewer 3: nit: maybe it would be good to specify that `cloud-readiness` is an example, something like:

> Run an analysis with a specified target technology, ie `cloud-readiness`:
